File: app/src/main/java/com/example/pictureme/models/Picme.java
package com.example.pictureme.models;

import com.google.firebase.firestore.DocumentId;

public class Picme {
    private String feeling;
    private boolean isFoodPic;

    public Picme() {
    }

    public Picme(String feeling, boolean isFoodPic) {
        this.feeling = feeling;
        this.isFoodPic = isFoodPic;
    }

    public String getFeeling() {
        return feeling;
    }

    public void setFeeling(String feeling) {
        this.feeling = feeling;
    }

    public boolean getIsFoodPic() {
        return isFoodPic;
    }

    public void setIsFoodPic(boolean isFoodPic) {
        this.isFoodPic = isFoodPic;
    }

    @Override
    public String toString() {
        return "Picme{" +
                "feeling='" + feeling + '\'' +
                ", isFoodPic=" + isFoodPic +
                '}';
    }
}

File: app/src/main/java/com/example/pictureme/data/models/Picme.kt
package com.example.pictureme.data.models

import com.google.firebase.Timestamp
import com.google.firebase.firestore.*
import java.io.File
import java.io.Serializable
import java.util.*

data class Picme(
    @DocumentId
    var id: String?,
    var imagePath: String?,
    var createdAt: Timestamp?,
    var location: GeoPoint?,
    @Exclude
    var creator: User?,
    @Exclude
    var feeling: Feeling?,
    @Exclude
    var friends: List<User>
) : Serializable {
    constructor() : this(null, null, null, null, null, null, emptyList())
}